File: ai-robot/src/main/java/za/co/sww/rwars/airobot/model/Robot.java
package za.co.sww.rwars.airobot.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Robot(
        String id,
        String name,
        String battleId,
        int positionX,
        int positionY,
        String direction,
        String status,
        int targetBlocks,
        int blocksRemaining,
        int hitPoints,
        int maxHitPoints
) {}

File: robo-demo/src/main/kotlin/za/co/sww/rwars/robodemo/model/Robot.kt
package za.co.sww.rwars.robodemo.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

/**
 * Represents a robot in the Robot Wars game.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class Robot(
    val id: String,
    val name: String,
    val battleId: String,
    val positionX: Int = 0,
    val positionY: Int = 0,
    val direction: String = "NORTH",
    val status: String = "IDLE",
    val blocksRemaining: Int = 0,
    val targetBlocks: Int = 0,
)
